sphinxcontrib/needs/api/need.py

```diff
@@ -370,7 +373,13 @@ def _read_in_links(links_string):
     # Get links
     links = []
     if links_string:
-        for link in re.split(";|,", links_string):
+        # Check if links_string is really a string, otherwise it will be a list, which can be used
+        # without modifications
+        if isinstance(links_string, str):
+            link_list = re.split(";|,", links_string)
+        else:
+            link_list = links_string
+        for link in link_list:
```

Comment:
do you want to add a type annotation to this function?

```python
def _read_in_links(links_string: Union[str, List[str]]):
    ...
```
?

Reply:
Yes, will add it.
Sorry I'm not so familiar with type annotations (till now) :)